### PR TITLE
feat(ScenarioSelector): adjust and improve styles

### PR DESCRIPTION
**Pull request by ghost**
Addresses the style part of #109

### CODE REVIEW

File: gui/src/ScenarioSelector.tsx [hunk 1]
```diff
@@ -177,43 +241,98 @@ const IconButtonSx = {
...
-        <IconButton data-id={scenarioId} onClick={openEditDialog}>
-          <EditOutlined fontSize="small" color="primary" />
+    <Grid
+      container
+      alignItems="center"
+      direction="row"
+      flexWrap="nowrap"
+      spacing={1}
+    >
+      <Grid item xs sx={{ display: "flex", alignItems: "center", gap: 1 }}>
```
Comment (FredLL-Avaiga) [inline gui/src/ScenarioSelector.tsx]: set the sx as a const outside the component

File: gui/src/ScenarioSelector.tsx [hunk 2]
```diff
@@ -177,43 +241,98 @@ const IconButtonSx = {
...
+        ) : (
+          <ScenarioIcon fontSize="small" color="primary" />
+        )}
+        {label}
+      </Grid>
+      <Grid item xs="auto">
+        <IconButton
+          data-id={scenarioId}
+          onClick={openEditDialog}
+          sx={{
```
Comment (FredLL-Avaiga) [inline gui/src/ScenarioSelector.tsx]: sx as const  with useMemo in the component

File: gui/src/ScenarioSelector.tsx [hunk 3]
```diff
@@ -177,43 +241,98 @@ const IconButtonSx = {
...
-              <Badge badgeContent={<FlagOutlined sx={FlagSx} />} color="primary" anchorOrigin={BadgePos} sx={BadgeSx}>
-                <ScenarioNodesContent scenarioId={id} label={label} openEditDialog={openEditDialog} />
-              </Badge>
-            ) : (
-              <ScenarioNodesContent scenarioId={id} label={label} openEditDialog={openEditDialog} />
-            )
+            <ScenarioNodesContent
+              scenarioId={id}
+              label={label}
+              isPrimary={showPrimary ? primary : undefined}
```
Comment (FredLL-Avaiga) [inline gui/src/ScenarioSelector.tsx]: showPrimary ? primary : undefined => showPrimary && primary

File: gui/src/ScenarioSelector.tsx [hunk 4]
```diff
@@ -310,7 +459,10 @@ const ScenarioEditDialog = ({ scenario, submit, open, actionEdit, configs, close
                           ))
                         : null}
                     </Select>
-                    <FormHelperText error={!!form.errors.config && form.touched.config} sx={{ pl: 12 }}>
+                    <FormHelperText
+                      error={!!form.errors.config && form.touched.config}
+                      sx={{ pl: 12 }}
```
Comment (FredLL-Avaiga) [inline gui/src/ScenarioSelector.tsx]: sx as const outside the component

File: gui/src/ScenarioSelector.tsx [hunk 5]
```diff
@@ -344,30 +498,88 @@ const ScenarioEditDialog = ({ scenario, submit, open, actionEdit, configs, close
...
+                          data-idx={index}
+                          onChange={updatePropertyField}
+                        />
                       </Grid>
-                      <Grid item xs={2}>
-                        <Button variant="outlined" component="label" data-id={item.id} onClick={propertyDelete}>
+                      <Grid
+                        item
+                        xs="auto"
+                        sx={{ display: "flex", alignItems: "flex-end" }}
```
Comment (FredLL-Avaiga) [inline gui/src/ScenarioSelector.tsx]: sx as const

File: gui/src/ScenarioSelector.tsx [hunk 6]
```diff
@@ -344,30 +498,88 @@ const ScenarioEditDialog = ({ scenario, submit, open, actionEdit, configs, close
...
+                    label="Value"
+                    variant="outlined"
+                  />
                 </Grid>
-                <Grid item xs={2}>
-                  <Button variant="outlined" component="label" onClick={propertyAdd} disabled={!newProp.key || !newProp.value}>
+                <Grid
+                  item
+                  xs="auto"
+                  sx={{ display: "flex", alignItems: "flex-end" }}
```
Comment (FredLL-Avaiga) [inline gui/src/ScenarioSelector.tsx]: sx as const

File: gui/src/ScenarioSelector.tsx [hunk 7]
```diff
@@ -463,51 +719,103 @@ const ScenarioSelector = (props: ScenarioSelectorProps) => {
...
                     {displayCycles ? (
                       nodeType === NodeType.CYCLE ? (
-                        <TreeItem key={id} nodeId={id} label={label} sx={CycleSx} data-cycle>
-                          <ScenarioNodes scenarios={scenarios} showPrimary={showPrimaryFlag} openEditDialog={openEditDialog} />
+                        <TreeItem
+                          key={id}
+                          nodeId={id}
+                          label={
+                            <Box
+                              sx={{
```
Comment (FredLL-Avaiga) [inline gui/src/ScenarioSelector.tsx]: sx as const

File: gui/src/icons/cycle-icon.component.tsx
Comment (FredLL-Avaiga) [inline gui/src/icons/cycle-icon.component.tsx:1]: naming => CycleIcon.tsx
Or maybe even Cycle.tsx as we're in icons dir

File: gui/src/icons/cycle-icon.component.tsx [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import { FC } from "react";
+import { SvgIcon, SvgIconProps } from "@mui/material";
+
+export const CycleIcon: FC<SvgIconProps> = (props) => (
```
Comment (FredLL-Avaiga) [inline gui/src/icons/cycle-icon.component.tsx]: consistency => export const CycleIcon = (props: SvgIconProps) => (

**Review by FredLL-Avaiga — DISMISSED** [review]
Tx @JaufreyTpy 
A few comments on externalizing sx.
And naming of icons (that I didn't expect you to do 👍 )

File: gui/src/ScenarioSelector.tsx (hunk 4 above)
Comment (ghost) [reply gui/src/ScenarioSelector.tsx]: Done

File: gui/src/ScenarioSelector.tsx (hunk 5 above)
Comment (ghost) [reply gui/src/ScenarioSelector.tsx]: Unnecessary after my intervention. Removed it.

File: gui/src/ScenarioSelector.tsx (hunk 6 above)
Comment (ghost) [reply gui/src/ScenarioSelector.tsx]: Same as above

File: gui/src/ScenarioSelector.tsx [hunk 9]
```diff
@@ -46,6 +53,8 @@ import {
   useDispatchRequestUpdateOnFirstRender,
   createSendUpdateAction,
 } from "taipy-gui";
+import { Cycle, Scenario } from "./icons";
+import { alpha, useTheme } from "@mui/material";
```
Comment (FredLL-Avaiga) [inline gui/src/ScenarioSelector.tsx]: sort import (external first)

File: gui/src/ScenarioSelector.tsx [hunk 10]
```diff
@@ -177,43 +248,105 @@ const IconButtonSx = {
   p: 0,
 };
 
-const ScenarioNodesContent = ({ scenarioId, label, openEditDialog }: ScenarioNodesContentProps) => {
+const ScenarioNodesContent = ({
```
Comment (FredLL-Avaiga) [inline gui/src/ScenarioSelector.tsx]: rename as ScenarioContent ?

File: gui/src/ScenarioSelector.tsx [hunk 11]
```diff
@@ -177,43 +248,105 @@ const IconButtonSx = {
...
+  scenarioId,
+  label,
+  isPrimary,
+  openEditDialog,
+}: ScenarioNodesContentProps) => {
+  const theme = useTheme();
+
+  const tinyEditIconButtonSx = React.useMemo(
+    () => ({
```
Comment (FredLL-Avaiga) [inline gui/src/ScenarioSelector.tsx]: would it be better to integrate tinyIconButtonSx here ?

**Review by FredLL-Avaiga — DISMISSED** [review]
minor changes

File: gui/src/ScenarioSelector.tsx (hunk 10 above)
Comment (FredLL-Avaiga) [reply gui/src/ScenarioSelector.tsx]: ScenarioItem

**Review by FabienLelaquais — APPROVED** [review]
👌

**Pull request merged**